keras/callbacks.py

```diff
@@ -694,13 +703,16 @@ def on_epoch_end(self, epoch, logs=None):
             if epoch % self.histogram_freq == 0:
                 # TODO: implement batched calls to sess.run
                 # (current call will likely go OOM on GPU)
+
+                val_data = self.validation_data
+                tensors = self.model.inputs + \
+                    self.model.targets + \
+                    self.model.sample_weights
+
                 if self.model.uses_learning_phase:
-                    cut_v_data = len(self.model.inputs)
-                    val_data = self.validation_data[:cut_v_data] + [0]
-                    tensors = self.model.inputs + [K.learning_phase()]
-                else:
-                    val_data = self.validation_data
-                    tensors = self.model.inputs
+                    tensors += [K.learning_phase()]
+
+                assert len(val_data) == len(tensors)
```

Comment:
Just sanity check. Tensorflow throws "you must feed a value" type of errors which are more difficult to follow, if this assert is not satisfied. So I thought it's more explicit to check it here.